### DISCUSSION TITLE
Items/Objects/Heroes

### DISCUSSION THREAD

**Discussion by AlexJ2017** [Programmiermethoden/Dungeon#8] [Ideas]
# Items und Objekte
Auflistung von möglichen Items und Objekten, die für den Dungeon verwendet werden sollen:

### Hebel, Schalter, Druckplatte
- Auslöser für verschiedene Aktionen:
  - Auslösen von Fallen
  - Öffnen von Türen
  - …
- Toggle oder einmaliges Einschalten

### Schlüssel
- Item zum Aufsammeln
- Zugeordnet zu einer oder mehreren Türen/Truhen

### Truhe
- Interagierbar (Öffnen mit oder ohne Schlüssel/Code)
- Inventar für Truheninhalt
  - Automatisches Inventar durch Loottables (levelabhängig)
  - Inventar durch Lehrenden festgelegt (Questitems)

### Tür
- Interagierbar (Öffnen mit oder ohne Schlüssel/Code)
- Kann vom Lehrenden verwendet werden um Abschnitte des Dungeons zu versperren, bevor festgelegte Bedingungen erfüllt wurden

### Falle
- Ausgelöst durch eine Aktion des Spielers oder Questfortschritt
- Schaden an Spielern (oder und zusätzlich an Gegnern)
- Rollender Fels:
  - Fällt von oben in den Dungeon und rollt in eine festgelegte Anfangsrichtung
  - Ändert bei Kontakt mit Wänden die Richtung
  - Zerfällt nach Anzahl zurückgelegter Tiles und oder Anzahl der Wandkontakte
- Pfeilschuss Anlage:
  - Schießt einen oder mehrere Pfeile in eine festgelegte Himmelsrichtung
- Stachelfalle:
  - Fährt Stacheln aus dem Boden aus
  - Variante A: regelmäßiges ein- und ausfahren
  - Variante B: ausgelöst durch eine Aktion oder Quest (vorher unsichtbar)

### Heil-/Mana-Brunnen
- Heilt den Helden/Füllt Mana des Helden wieder auf
- Einmalige Aktion bei Interaktion

### Munitionskisten
- Item zum Aufsammeln
- Füllt beim Aufsammeln Munitionsvorrat wieder auf
- Bogen (Pfeile), Gewehre (Patronen), …

### Teleporter
- Jeweils zwei verbinden Abschnitte des Dungeons miteinander
- Teleportieren den Spieler bei Interaktion

### Trank
- Item zum Aufsammeln
- Varianten: Heilung, Mana, temporäre/permanente Buffs

# Helden
Mögliche Heldenklassen, die den Spielern zur Auswahl stehen:
### Ritter
- Nahkämpfer mit Schwert (und Schild)

### (Bogen)schütze
- Fernkämpfer mit Bogen oder Armbrust
- Tarnfähigkeit zum Vorbeischleichen an Gegnern

### Magier
- Fernkämpfer mit Magischen Geschossen
- Heilzauber
- Schockwellen Zauber zum Wegstoßen von Gegnern (und rollenden Felsen)

### Mönch
- Nahkämpfer mit Fäusten oder Stab

### Drache
- Nahkampf mit Klauen und Fernkampf mit Feuerball
- (Natürliche Regeneration von Leben und Mana)

**Comment by AMatutat** [comment]
@AHeinisch @Lena241 
Hier haben wir auch schon ein paar Ideen für das Spiel gesammelt. ggf. könnt ihr hier nochmal was raus ziehen.
